File: move2alf-pipeline/src/main/scala/eu/xenit/move2alf/pipeline/actions/ReceivingAction.scala
package eu.xenit.move2alf.pipeline.actions

import eu.xenit.move2alf.pipeline.{M2AMessage, AbstractMessage}

/**
 * Created with IntelliJ IDEA.
 * User: thijs
 * Date: 3/6/13
 * Time: 8:06 PM
 * To change this template use File | Settings | File Templates.
 */
trait ReceivingAction[T <: AbstractMessage] extends AbstractAction {

  override def receive = {
    case M2AMessage(message) => execute(message.asInstanceOf[T])
    case s => super.receive(s)
  }

  protected def execute(message: T)

}

File: move2alf-pipeline/src/main/java/eu/xenit/move2alf/pipeline/actions/ReceivingAction.java
package eu.xenit.move2alf.pipeline.actions;

/**
 * Created with IntelliJ IDEA.
 * User: thijs
 * Date: 4/25/13
 * Time: 12:05 PM
 * To change this template use File | Settings | File Templates.
 */
public interface ReceivingAction<T> extends Action{

    public void execute(T message);

}
